queue/management/commands/queue_ungraded_submissions.py

```diff
@@ -0,0 +1,117 @@
+        )
+        parser.add_argument(
+            '--pull-time-start',
+            dest='pull_time_start',
+            default=None,
+            help='Submission pull_time range start (ISO-8601 formatted - 2017-01-01T00:00:00Z, et. al.)',
+        )
+        parser.add_argument(
+            '--pull-time-end',
+            dest='pull_time_end',
+            default=None,
+            help='Submission pull_time range end (ISO-8601 formatted - 2017-01-01T00:00:00Z, et. al.)',
+        )
+        parser.add_argument(
+            '--ignore-failures',
+            action='store_true',
+            dest='ignore_failures',
+            default=False,
+            help='Requeue the submissions even if their num_failures is greater than the max in settings',
+        )
+        parser.add_argument(
+            '--echo',
+            action='store_true',
+            dest='echo',
+            default=False,
+            help='Echo the submissions that will be queued for grading (instead of actually queueing them)',
+        )
+
+    def handle(self, *args, **options):
+        query_exclude_params = dict(
+            lms_ack=1,
+            retired=1
+        )
+        query_param_dict = {}
+        query_param_list = []
+        if options['queue_names']:
+            query_param_dict['queue_name__in'] = options['queue_names'].split(',')
+        if options['submission_ids']:
+            query_param_dict['id__in'] = options['submission_ids'].split(',')
+        if options['pull_time_start']:
+            query_param_dict['pull_time__gte'] = parse_iso_8601_string(options['pull_time_start'])
+        if options['pull_time_end']:
+            query_param_dict['pull_time__lte'] = parse_iso_8601_string(options['pull_time_end'])
+            query_exclude_params['pull_time'] = None
+        else:
+            query_param_list.append(
+                Q(pull_time__lte=timezone.now() - timedelta(seconds=settings.PULLED_SUBMISSION_TIMEOUT)) |
+                Q(pull_time=None)
+            )
+        if not options['ignore_failures']:
+            query_param_dict['num_failures__lt'] = settings.MAX_NUMBER_OF_FAILURES
+
+        submission_qset = (
+            Submission.objects
+            .filter(*query_param_list, **query_param_dict)
+            .exclude(**query_exclude_params)
+            .order_by('-push_time')
+        )
+        if options['echo']:
+            pp = pprint.PrettyPrinter(indent=2)
+            for submission in submission_qset.values(*self.ECHO_PROPERTIES)[0:self.ECHO_LIMIT]:
+                self.stdout.write(pp.pformat(submission))
+            num_submissions = submission_qset.count()
+            self.stdout.write("\nMatching submission count: {}".format(num_submissions))
+            if num_submissions > self.ECHO_LIMIT:
+                submission_ids = submission_qset.values_list('id', flat=True)
+                self.stdout.write("\nIDs: {}\n\n".format(','.join(map(str, submission_ids))))
```

Comment:
I'm not sure if edX supports Python 2.6, but if they do then we will need to add explicit index values to the format string (e.g. `'foo {0}'.format('bar')`) if we want to get this merged upstream.